### PR TITLE
Added `ResourceDataSync` resource support 

### PR DESCRIPTION
**Pull request by rushmash91**
Issue #, if available:

Description of changes:
- Api does not support tags
- no 'type_' issues

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_resourcedatasync.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        k8s.patch_custom_resource(reference, update_data)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True", wait_periods=10)
+
+        updated_cr = k8s.get_resource(reference)
+        assert updated_cr is not None
+        assert 'sourceRegions' in updated_cr["spec"]["syncSource"]
+        assert updated_cr["spec"]["syncSource"]["sourceRegions"] == update_data["spec"]["syncSource"]["sourceRegions"]
```
Comment (a-hilaly) [inline test/e2e/tests/test_resourcedatasync.py:76]: Can we also call the AWS api and verify this information?

**Review by a-hilaly — COMMENTED** [review]
Nice! thank you @rushmash91 !

File: test/e2e/tests/test_resourcedatasync.py (hunk 1 above)
Comment (rushmash91) [reply test/e2e/tests/test_resourcedatasync.py]: added aws api asserts

**Review by a-hilaly — APPROVED** [review]
Nice, good job @rushmash91 !!
/lgtm

File: test/e2e/tests/test_resourcedatasync.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+
+        ssm_client = boto3.client('ssm')
```
Comment (a-hilaly) [inline test/e2e/tests/test_resourcedatasync.py:80]: nit: we define AWS clients usually in a different file https://github.com/aws-controllers-k8s/eks-controller/blob/main/test/e2e/conftest.py#L52-L54

**Pull request merged**